rider/buildSrc/src/main/kotlin/com/jetbrains/rider/plugins/gradle/tasks/GenerateBuildPropsTask.kt

```diff
@@ -0,0 +1,70 @@
+package com.jetbrains.rider.plugins.gradle.tasks
+
+import com.jetbrains.rider.plugins.gradle.buildServer.buildServer
+import org.gradle.api.DefaultTask
+import org.gradle.api.GradleException
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.OutputFile
+import org.gradle.api.tasks.TaskAction
+import java.io.File
+import java.util.regex.Pattern
+
+open class GenerateBuildPropsTask: DefaultTask() {
+
+    @Input
+    var packagesDirectory: Any? = null
+
+    var packageName: String = ""
+    var msBuildParameter: String = ""
+
+    @Input
+    var packageVersion = project.providers.provider({ parsePackageVersion() })
+
+    @OutputFile
+    var propsFile = project.providers.provider({ File("${project.projectDir}/../resharper/$packageName.generated.props") })
+
+    @TaskAction
+    fun generate() {
+        project.buildServer.progress("Generating build.props for $packageName")
+
+        val version = packageVersion.get()
+        logger.info("$msBuildParameter=$version")
+
+        project.file(propsFile).writeText("""<Project>
+  <PropertyGroup>
+    <$msBuildParameter>[$version]</$msBuildParameter>
+  </PropertyGroup>
+</Project>
+""")
+    }
+
+    private var parsedPackageVersion: String? = null
+
+    private fun parsePackageVersion(): String? {
+        if (parsedPackageVersion != null) return parsedPackageVersion
+
+        if (packagesDirectory == null) {
```

Comment:
is there no convenient Syntax for the asserts like in groovy? e.g. `.existsDirectory()`

Reply:
Because of the type coercion (see below), you can't really assert it, you can only pass it to `project.file()` to convert it to a file and that has to be non-null.